codalab/worker/bundle_service_client.py

```diff
@@ -23,14 +23,15 @@ def wrapper(*args, **kwargs):
             except urllib.error.HTTPError as e:
                 try:
                     client_error = e.read()
+                    error_code = e.code.decode() if isinstance(e.code, bytes) else e.code
```

Comment:
What are possible values of `e.code` and when is it bytes?

Reply:
Can we assume the `decode()` always succeeds?

Reply:
This bug didn't happen very often. Last time I saw it was on production, but didn't save it. I just found the same one when I was testing locally a moment ago. It looks like this:
```
stanford@Stanfords-MacBook-Pro codalab-worksheets % cl-worker --server="http://localhost" --tag stan
Username: codalab
Password: 
2020-02-07 11:05:24,172 Loaded 0 dependencies, 0 paths from cache.
2020-02-07 11:05:26,464 Cannot initialize NVIDIA runtime, no GPU support: Problem getting NVIDIA devices: 400 Client Error: Bad Request ("Unknown runtime specified nvidia")
2020-02-07 11:05:26,794 Worker started!
2020-02-07 11:05:26,794 Starting docker image manager
2020-02-07 11:05:26,794 Starting local dependency manager
2020-02-07 11:05:29,818 Connected! Successful check in!
2020-02-07 11:13:50,341 Downloading tag information for codalab/default-cpu:latest
^[[ATraceback (most recent call last):
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/bundle_service_client.py", line 18, in wrapper
    return f(*args, **kwargs)
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/bundle_service_client.py", line 122, in checkin
    'POST', self._worker_url_prefix(worker_id) + '/checkin', data=request_data
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/rest_client.py", line 98, in _make_request
    with closing(urllib.request.urlopen(request)) as response:
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/worker.py", line 171, in start
    self.checkin()
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/worker.py", line 244, in checkin
    response = self.bundle_service.checkin(self.id, request)
  File "/Users/stanford/PycharmProjects/codalab-worksheets/codalab/worker/bundle_service_client.py", line 33, in wrapper
    message + ': ' + http.client.responses[e.code] + ' - ' + client_error,
TypeError: can only concatenate str (not "bytes") to str
2020-02-07 11:14:24,541 Sleeping for 1 hour due to exception...please help me!
```